src/videojs.ads.js

```diff
@@ -477,11 +460,33 @@ var
             }
           },
           'content-playback': {
+            enter: function() {
+              player.trigger({
+                type: 'contentplayback',
+                triggerevent: fsm.triggerevent
+              });
+              // Make sure that the cancelPlayTimeout is cleared
+              if (player.ads.cancelPlayTimeout) {
+                clearImmediate(player.ads.cancelPlayTimeout);
+                player.ads.cancelPlayTimeout = null;
+                // Trigger playback if play was canceled and the player is paused.
+                if (player.paused()) {
+                  player.play();
+                }
+              }
+            },
             events: {
+              // in the case of a timeout, adsready might come in late.
+              'adsready': function() {
+                if (player.paused()) {
+                  this.state = 'ads-ready';
+                } else {
+                  this.state = 'preroll?';
```

Comment:
I ended up making this change as you suggested.  This basically means that the ad-integrations are on their own in this scenario though (adsready not triggering until content has already started to play).  I concluded that was reasonable because this scenario is something each integration would probably want to deal with differently anyway.  Some might want to choose to halt the content playback play the preroll and resume,  some might want to play the preroll and restart the content, and others might want to just skip the preroll all-together.